src/hooks/usePropsGetters/useOpenTextInputProps.ts

```diff
@@ -0,0 +1,62 @@
+import { useState, useCallback, useEffect } from 'react';
+import { AnswerInputState, GetOpenTextInputProps, Question, QuizEventsReturn } from '../../types';
+
+export default function useOpenTextInputProps(
+  setQuizAnswers: QuizEventsReturn.QuizAnswerChanged,
+  nextQuestion: QuizEventsReturn.NextQuestion,
+  currentQuestionData?: Question,
+  answerInputs?: AnswerInputState
+): GetOpenTextInputProps {
+  const [input, setInput] = useState('');
+
+  const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInput(e.target.value);
+  };
+
+  const onKeyDownHandler = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    const { key } = e;
+
+    if (key === 'Enter') {
+      if (input && currentQuestionData?.type === 'open') {
+        nextQuestion();
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (currentQuestionData?.type === 'open') {
+      setQuizAnswers(input);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [input, currentQuestionData]);
```

Comment:
React Question: Would changing this to `currentQuestionData.type` help here?

Reply:
yup, changing this to `currentQuestionData.type` would cause this to effect to run only when the value of  `currentQuestionData.type`  changes as opposed to every time that `currentQuestionData` references a new object

Reply:
It doesn't really matter much here since when type change is when currentQuestionData change they happen together

Reply:
But wouldn't that allow us to remove `react-hooks/exhaustive-deps`?

Reply:
yup and the underlying reasoning for WHY this'll make the linter happy is that the linter is enforcing the pattern that will help to avoid difficult to diagnose performance issue that can result from using objects in dependency arrays. 

We should only use primitives in dependency arrays